File: PRACTICA PRIMER PARCIAL/Ejercicio25/Python/DivisoresPares.py
n = int(input("Ingrese la cantidad de números en el lote: "))
numeros = []
for i in range(n):
    numero = int(input("Ingrese el número " + str(i+1) + ": "))
    numeros.append(numero)

divisores_pares = []
for num in numeros:
    if num % 2 == 0:
        divisores = [i for i in range(1, num+1) if num % i == 0]
        divisores_pares.append((num, divisores))

for par in divisores_pares:
    numero = par[0]
    divisores = par[1]
    print("Divisores pares de", numero, ":", divisores)

File: PRACTICA PRIMER PARCIAL/Ejercicio25/Java/DivisoresPares.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class DivisoresPares {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese la cantidad de números en el lote: ");
        int n = scanner.nextInt();

        List<Integer> numeros = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            System.out.print("Ingrese el número " + (i+1) + ": ");
            int numero = scanner.nextInt();
            numeros.add(numero);
        }

        List<Pair<Integer, List<Integer>>> divisoresPares = new ArrayList<>();
        for (int num : numeros) {
            if (num % 2 == 0) {
                List<Integer> divisores = new ArrayList<>();
                for (int i = 1; i <= num; i++) {
                    if (num % i == 0) {
                        divisores.add(i);
                    }
                }
                divisoresPares.add(new Pair<>(num, divisores));
            }
        }

        for (Pair<Integer, List<Integer>> par : divisoresPares) {
            int numero = par.getKey();
            List<Integer> divisores = par.getValue();
            System.out.println("Divisores pares de " + numero + ": " + divisores);
        }
    }

    static class Pair<K, V> {
        private K key;
        private V value;

        public Pair(K key, V value) {
            this.key = key;
            this.value = value;
        }

        public K getKey() {
            return key;
        }

        public V getValue() {
            return value;
        }
    }
}